### PR TITLE
Exercises

### PR DESCRIPTION
**Pull request by qKarpej**

### CODE REVIEW

File: exercises/01_trim_right/main.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+func TrimRight(s string) string {
+	for s != "" && InSet(s[len(s)-1], trimSet) {
+		s = s[:len(s)-1]
+	}
+	return s
+}
+
+func main() {
+	fmt.Print("Enter a string: ")
```
Comment (yarcat) [inline exercises/01_trim_right/main.go:28]: nit: Please drop this confusing string. Input string `Enter a string: Enter text: ` looks weird

File: exercises/01_trim_right/main_test.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import "testing"
+
+func TestTrimRight(t *testing.T) {
+	for _, tc := range []struct {
+		name  string
+		input string
+		want  string
+	}{
+		{"empty", "", ""},
+		{"nothing to trim", "abc", "abc"},
```
Comment (yarcat) [inline exercises/01_trim_right/main_test.go:12]: nit: I also recomment to add the following two/three tests:
- `"a"` just a single non-space
- `" "` just a single space
- `"a "` a single non-space + space
Often there are bugs (your implemnetation works just fine though) that work for general cases, but fail for some specific corner-cases. And these are good simple tests to run.

File: exercises/03_count_byte/main_test.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package main
+
+import "testing"
+
+func TestCountByte(t *testing.T) {
+	for _, tc := range []struct {
+		name   string
+		input  string
+		input2 byte
```
Comment (yarcat) [inline exercises/03_count_byte/main_test.go:9]: nit: I recommend name the same way you name arguments.

File: exercises/03_count_byte/main_test.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+		want   int
+	}{
+		{"letter is not repeated", "mango", byte('l'), 0},
```
Comment (yarcat) [inline exercises/03_count_byte/main_test.go:12]: nit: You don't have to cast `'l'` to byte. `'l'` is an untyped constant, and Go will automatically make a byte out of it.

File: exercises/03_count_byte/main_test.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+		{"letter is not repeated", "mango", byte('a'), 1},
+		{"letter is repeated twice", "hello", byte('p'), 0},
+		{"letter is repeated twice", "hello", byte('e'), 1},
+		{"letter is repeated twice", "hello", byte('l'), 2},
+		{"Latvian letters", "abčdē", 0xc4, 2},
```
Comment (yarcat) [inline exercises/03_count_byte/main_test.go:17]: Emmm... Nothing is wrong here? U+00C4 is a `'Ä'` letter. Do you see any letter like that in the input string??? You shouldn't be passing bytes above 127, since those require unicode handling!

File: exercises/04_count_rune/main.go [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+)
+
+func CountRune(s string, b rune) int {
+	var count int
+	s1 := []rune(s)
```
Comment (yarcat) [inline exercises/04_count_rune/main.go:11]: nit:

1) You don't need any temporary variables for this
```go
for _, c := range []rune(s) { ... }
```

2) You don't even have to cast, since `for-range` defaults to `rune`:
```go
for _, c := range s {
  if c == b { count++ }
}
```

File: exercises/04_count_rune/main.go [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+	for i := 0; i < len(s); i++ {
```
Comment (yarcat) [inline exercises/04_count_rune/main.go:12]: Your program should be failing for any unicode examples since `len(s)` is longer than `len([]rune(s))` for strings that contain unicode characters

Comment (yarcat) [reply exercises/04_count_rune/main.go:12]: Yup, just confirmed that by running your tests
